### PR TITLE
Add virtualization in sites table

### PR DESCRIPTION
**Pull request by K-Markopoulos**
The goal is to improve the performance of the sites table in `/map` route, by adding virtualization. Currently we do not paginate the sites and when we fetch a few thousands sites, the UI is slow and may freeze.

- Add `react-window` package which handles virtualization.
- Add `VirtualTable` component "forked" from this [sandbox](https://codesandbox.io/p/sandbox/react-window-with-table-elements-d861o), as `react-window` only works for lists and we need to make it work with tables.
- Rename `SiteTableBody` -> `SiteTable`, it now renders both header (received from props) and body.
- Rename `SiteTable` -> `SiteTableContainer`

### CODE REVIEW

**Comment by ericboucher** [comment]
Nice work!

- I noticed a small visual glitch when scrolling down, the header moves slightly creating a visual glitch
- clicking on a reef in the table is super slow on the staging env, but this might be due to the high number of points on the map and not the table? If it's the table then we should investigate.
- Sorry I didn't think of that earlier but another option would be to use pagination instead. Do you have any thoughts on pagination vs virtualization in this case? Noting that a short-term goal is to put filters more at the center of the usage of the map "à la" Airbnb.

**Comment by K-Markopoulos** [comment]
- The visual glitch is unfortunately present because the items are absolute positioned on scroll. At first look I had not luck mitigating this issue, we will need to investigate it further.
- The slowness is most likely because of the map, if there is no interaction with the map the page and the table is very fast. I tried to remove the map and the item clicking does cause any issue.
- Pagination (server side) seems like a better solution to avoid fetching thousands of items. Virtualization is a quicker solution to avoid rendering all of them, if we are fetching thousands, let's not render them all at once. But adding pagination here might be tricky because the paginated items have to be in sync with the visible items in the map. There are some edge cases we need to figure out like, what happens when the map is completely zoomed out, if the map is fetching all items to display them then they are not paginated and we don't have a specific page to show. It can be achieved but it needs more thought.

File: packages/website/src/routes/HomeMap/SiteTableContainer/table.tsx [hunk 1]
```diff
@@ -288,28 +336,30 @@
       },
     },
     tableRow: {
+      position: 'static !important' as 'static',
       cursor: 'pointer',
       borderTop: `1px solid ${theme.palette.grey['300']}`,
     },
   });
 
 type SiteTableBodyIncomingProps = {
```
[the remark this answers is not in the document]
Comment (K-Markopoulos) [reply packages/website/src/routes/HomeMap/SiteTableContainer/table.tsx:345]: Right, nice catch :fishing_pole_and_fish: !

**Comment by ericboucher** [comment]
> @ericboucher
> 
> * The visual glitch is unfortunately present because the items are absolute positioned on scroll. At first look I had not luck mitigating this issue, we will need to investigate it further.
> * The slowness is most likely because of the map, if there is no interaction with the map the page and the table is very fast. I tried to remove the map and the item clicking does cause any issue.
> * Pagination (server side) seems like a better solution to avoid fetching thousands of items. Virtualization is a quicker solution to avoid rendering all of them, if we are fetching thousands, let's not render them all at once. But adding pagination here might be tricky because the paginated items have to be in sync with the visible items in the map. There are some edge cases we need to figure out like, what happens when the map is completely zoomed out, if the map is fetching all items to display them then they are not paginated and we don't have a specific page to show. It can be achieved but it needs more thought.

**Comment by ericboucher** [comment]
Closing for now in favor of pagination but let's keep this in mind !

**Pull request closed without merging**